test/integration/span/spanlinks_test.go

```diff
@@ -0,0 +1,65 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2023-present Datadog, Inc.
+
+package agent
+
+import (
+	"bytes"
+	"context"
+	"net/http"
+	"sync"
+	"testing"
+
+	"github.com/DataDog/datadog-agent/pkg/trace/agent"
+	"github.com/DataDog/datadog-agent/pkg/trace/config"
+	"github.com/DataDog/datadog-agent/pkg/trace/telemetry"
+	"github.com/DataDog/datadog-agent/pkg/trace/testutil"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestSpanLinks(t *testing.T) {
+	cfg := config.New()
+	cfg.Endpoints[0].APIKey = "test"
+	cfg.AgentVersion = "testVersion"
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	agnt := agent.NewAgent(
+		ctx,
+		cfg,
+		telemetry.NewNoopCollector())
+	agnt.Receiver.Start()
+
+	testTraces := testutil.RandomTrace(10, 20)
+	msgpBts, err := testTraces.MarshalMsg(nil)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	for _, endpoint := range []string{
+		"/v0.4/traces",
+		"/v0.7/traces",
+	} {
+		req, err := http.NewRequest("POST", "http://localhost:8126"+endpoint, bytes.NewReader(msgpBts))
+		if err != nil {
+			t.Fatal(err)
+		}
+		var wg sync.WaitGroup
```

Comment:
You need to call `wg.Wait()` later ([example](https://gobyexample.com/waitgroups))

Reply:
I'm not sure we'd need concurrency in the first place.

Reply:
Removed